packages/amplify_authenticator/lib/src/widgets/authenticator_banner.dart

```diff
@@ -79,4 +77,17 @@ extension on StatusType {
         return Icons.error;
     }
   }
+
+  ColorSwatch<int> get color {
```

Comment:
Wondered if there is a way to tie these colors to the theme and allow customization by devs?

Reply:
Yeah we need to make this so that customers can still set this via the theme. These colors seem like reasonable defaults, but customers need to be able to change them. You might be able to check to see if the theme has a color set and use this as a fallback.

The SnackBar foreground color is set to `theme.snackBarTheme.contentTextStyle?.color ?? theme.colorScheme.surface`. It is using `theme.colorScheme.surface` as a fallback. You could update it to to `theme.snackBarTheme.contentTextStyle?.color ?? type.color` so that it uses this value as a fallback, but still allows customers to theme it. That is for SnackBar, but the same could be applied to material banner.

We should (eventually) allow customers to provide a theme for each type (info, success, error) but that should be part of a larger theming API change. See https://github.com/aws-amplify/amplify-flutter/pull/1850/files#diff-fc4479f329ca30aeb3fc007095889bc66b798e3f77569534ca9bfcd1971bb216R83-R92

Whatever updates we make to material banner we should probably make to snack bar as well.

Do these colors look reasonable in light and dark mode (Does the text color invert? If so, is that what we want?)

Reply:
Great insight. Here are some options I've uncovered based on what you shared 

1) Tie the colors to the snackbar theme as suggested.

2) Use the material banner specific theme settings, eg. `theme.bannerTheme.backgroundColor`.

3) Tie the different type colors individually to theme settings like `theme.errorColor` & `theme.hintColor`

Re: light/dark mode - the colors do invert based on the mode and they look readable/reasonable to me. Now if this is what we want is probably a better question for a designer haha.

Reply:
Let's use the bannerTheme for MaterialBanner and the snackBarTheme for SnackBar 👍 . Also, I just realized I was mixing up background and foreground color. Obviously let's use the SnackBar / banner theme *background* to determine the background color, and user the theme's *foreground* color to determine the text/icon/etc. I mixed it up in the code snippet.

```dart
// SnackBar background
final backgroundColor = theme.snackBarTheme.backgroundColor ?? type.color;

// Banner background
final backgroundColor = theme.bannerTheme.backgroundColor ?? type.color;
```

Re: light/dark mode - Yeah, this is a bit of an opinion. Let's make sure it meets accessibility standards, and then try to use Material Design standards or defaults where possible to guide us.

Here is what Material has to say about [text on colored backgrounds](https://material.io/design/color/text-legibility.html#text-backgrounds).

> Black text is recommended for use on light backgrounds, and white text on dark backgrounds. If your app has both light and dark themes, make sure the text is available in a contrasting color against each theme.

You could probably interpret this multiple ways, but the way I interpret this is - Text color should be a function of background color. If your app has different background colors in light/dark mode, make sure your text responds to that. That is what flutter does by default. SnackBar has a different background color in light/dark mode, and the foreground color responds to that.

In our case, I think we are going to use red/blue/green as the background in both light and dark mode, so I don't think the foreground color needs to adapt. I would choose either white or black and use it as the (fallback) foreground color regardless of the theme.

So how do we know if we should use black or white? Material has a great interactive guide for this. You pick a Material theme (or define a custom one) and it will help you decide what color the text should be based on the color/size/weight. Here is the link for [Colors.red](https://material.io/resources/color/#!/?view.left=1&view.right=0&primary.color=F44336). The guide here says that both white and black text is fine if the text is "large" but will be illegible for "normal" text (tooltip shows the definition of large/normal text). In other terms, black text will always be legible on this background, but white will only be legible if the font is large/bold. So, I would choose black in this case. It happens to be the same for blue and green as well. If you chose indigo, it would be the opposite - white is always legible, and black is only legible if the text is large/bold.

And finally we have to consider if the user has set values for either foreground or background.

This is what I am thinking the logic should be. This is for SnackBar, but should be adaptable for banners.

```dart
  // foreground and background colors are based on material standards for text legibility
  // if they are not provided by the theme. See https://material.io/design/color/text-legibility.html#text-backgrounds

  // fallback on default color based on type.
  final backgroundColor = theme.snackBarTheme.backgroundColor ?? type.color;

  // If there is no background color provided by the theme,
  // use on Colors.black87 (High-emphasis) as fallback, otherwise 
  // set it to null to let flutter control the foreground.
  final hasCustomBackground = theme.snackBarTheme.backgroundColor != null;
  final foregroundColorFallBack = hasCustomBackground ? null : Colors.black87;

  // use contentTextStyle color if available, otherwise use fallback as defined above.
  final foregroundColor =
      theme.snackBarTheme.contentTextStyle?.color ?? foregroundColorFallBack;
```